src/smc-webapp/r_misc/index.tsx

```diff
@@ -14,22 +18,35 @@ export { CloseX2 } from "./close-x2";
 export { SimpleX } from "./simple-x";
 export { Saving } from "./saving";
 export { Spinner } from "./spinner";
-export { ErrorDisplay } from "./error-display";
+export const ErrorDisplay = is_frontend
+  ? require("./error-display").ErrorDisplay
+  : require("./error-display-backend").ErrorDisplay;
 export { SkinnyError } from "./skinny-error";
 export { SelectorInput } from "./selector-input";
 export { TextInput } from "./text-input";
-export { NumberInput } from "./number-input";
+export const NumberInput = is_frontend
+  ? require("./number-input").NumberInput
+  : require("./number-input-backend").NumberInput;
 export { LabeledRow } from "./labeled-row";
 export { TimeElapsed } from "./time-elapsed";
 export { SettingBox } from "./setting-box";
 export { ProfileIcon } from "./profile-icon";
-export { DropdownMenu, MenuItem, MenuDivider } from "./dropdown-menu";
+
+const dropdown_menu_module = is_frontend
+  ? require("./dropdown-menu")
+  : { DropdownMenu: Null, MenuItem: Null, MenuDivider: Null };
+export const { DropdownMenu, MenuItem, MenuDivider } = dropdown_menu_module;
 export { WindowedList } from "./windowed-list";
 export { UncommittedChanges } from "./uncommited-changes";
-export { DateTimePicker } from "./date-time-picker";
-export { PathLink } from "./path-link";
-export { HelpIcon } from "./help-icon";
-export { TimeAgo, TimeAgoElement, is_different_date } from "./time-ago";
+export const DateTimePicker = is_frontend
+  ? require("./date-time-picker").DateTimePicker
+  : Null;
+export const PathLink = is_frontend ? require("./path-link").PathLink : Null;
+export const HelpIcon = is_frontend ? require("./help-icon").HelpIcon : Null;
+const time_ago_module = is_frontend
```

Comment:
Does this approach break all the Typescript typing?  If so, it's not acceptable.

Reply:
I just did a test and in master (so without your changes), doing `<TimeAgo foo="bar" date={new Date()}>` gives a big type error (yeah!).  After checking out this branch the error disappears.  Thus unfortunately this approach is simply not going to work, since typing is extremely important.